src/output/flat.rs

```diff
@@ -217,14 +217,23 @@ impl<'module, W: io::Write> Context<'module, W> {
                 }
                 _ => panic!("expected variant"),
             },
-            Format::Union(branches) | Format::NondetUnion(branches) => match value {
-                Value::Variant(label, value) => {
-                    let (_, format) = branches.iter().find(|(l, _)| l == label).unwrap();
+            Format::UnionVariant(branches) | Format::UnionNondet(branches) => match value {
+                Value::Branch(index, value) => {
+                    let format = &branches[*index].1;
```

Comment:
Do we potentially want a sanity-check that `&branches[*index].0 == _label`, or is that perfunctory?